src/Group.php

```diff
@@ -4,53 +4,63 @@
 
 namespace Yiisoft\Router;
 
+use Attribute;
 use InvalidArgumentException;
 use RuntimeException;
-use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 
 use function in_array;
 
+#[Attribute(Attribute::TARGET_CLASS)]
 final class Group
 {
     /**
      * @var Group[]|Route[]
      */
-    private array $items = [];
-
+    private array $routes = [];
+    private bool $routesAdded = false;
+    private bool $middlewareAdded = false;
+    private array $builtMiddlewares = [];
     /**
-     * @var array[]|callable[]|string[]
+     * @var array|callable|string|null Middleware definition for CORS requests.
      */
-    private array $middlewareDefinitions = [];
-
+    private $corsMiddleware;
     /**
      * @var string[]
      */
     private array $hosts = [];
-    private ?string $namePrefix = null;
-    private bool $routesAdded = false;
-    private bool $middlewareAdded = false;
-    private array $disabledMiddlewareDefinitions = [];
-
     /**
-     * @var array|callable|string|null Middleware definition for CORS requests.
+     * @var array[]|callable[]|string[]
      */
-    private $corsMiddleware = null;
+    private array $middlewares = [];
 
-    private function __construct(private ?string $prefix = null, private ?MiddlewareDispatcher $dispatcher = null)
-    {
+    /**
+     * @param array $disabledMiddlewares Excludes middleware from being invoked when action is handled.
+     * It is useful to avoid invoking one of the parent group middleware for
+     * a certain route.
+     */
+    public function __construct(
+        private ?string $prefix = null,
+        array $middlewares = [],
+        array $hosts = [],
+        private ?string $namePrefix = null,
+        private array $disabledMiddlewares = [],
+        array|callable|string|null $corsMiddleware = null
+    ) {
+        $this->assertMiddlewares($middlewares);
+        $this->assertHosts($hosts);
+        $this->middlewares = $middlewares;
+        $this->hosts = $hosts;
+        $this->corsMiddleware = $corsMiddleware;
     }
 
     /**
      * Create a new group instance.
      *
      * @param string|null $prefix URL prefix to prepend to all routes of the group.
-     * @param MiddlewareDispatcher|null $dispatcher Middleware dispatcher to use for the group.
      */
-    public static function create(
-        ?string $prefix = null,
-        MiddlewareDispatcher $dispatcher = null
-    ): self {
-        return new self($prefix, $dispatcher);
+    public static function create(?string $prefix = null): self
```

Comment:
How about to eliminate this?

Reply:
Create group via `new Group` is non-comfortable in application configuration.

Reply:
Most case of usage groups required chain calls.

1) Now:  `Group::create('/xxx')->...`
2) Your suggestion: `(new Group('/xxx'))->...`

Brackets in second way is non-comfortable =)